internal/models/user.go

```diff
@@ -294,7 +294,16 @@ func (u *User) SetPassword(ctx context.Context, password string) error {
 		return err
 	}
 
-	u.EncryptedPassword = pw
+	if encrypt {
+		es, err := crypto.NewEncryptedString(u.ID.String(), []byte(pw), encryptionKeyID, encryptionKey)
+		if err != nil {
+			return err
+		}
+
+		u.EncryptedPassword = es.String()
+	} else {
+		u.EncryptedPassword = pw
+	}
```

Comment:
nit: would prefer to handle the default case always first 
```suggestion
        u.EncryptedPassword = pw
	if encrypt {
		es, err := crypto.NewEncryptedString(u.ID.String(), []byte(pw), encryptionKeyID, encryptionKey)
		if err != nil {
			return err
		}

		u.EncryptedPassword = es.String()
	}
```

Reply:
also how do we differentiate between a password and an encrypted one?

Reply:
See description. :D